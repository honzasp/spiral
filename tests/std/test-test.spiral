(program
  (import std)
  (import std.test)

  (test "success" (lambda (t)
    (assert t true)
    (assert t 42)
    (assert-not t false)
    (assert-eq t 20 20)
    (assert-near-eq t 0.1 2.34 2.28)
    (assert-eqv t "a" "a")
    (assert-equal t (cons "a" 10) (cons "a" 10))))
  (test "failure" (lambda (t)
    (assert t false)
    (assert t true)
    (assert-not t 42)
    (assert-eq t 10 10)
    (assert-eq t 10 20)
    (assert-near-eq t 0.1 10 11)
    (assert-eqv t "a" "b")
    (assert-equal t (cons "a" "b") (cons "b" "a")))))
