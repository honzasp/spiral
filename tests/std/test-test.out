test success ok
test failure failed:
  expected true
  expected false, got 42
  expected == 10 20
  expected near == 10 11
  expected eqv a b
  expected equal (a . b) (b . a)
